### PR TITLE
containerupdate: improved error messaging when the file system isn't writable

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @landism,

Please review the following commits I made in branch nicks/ch11396:

63fa3751e92f147ee790ed4b289951ff45823a66 (2021-02-16 18:20:57 -0500)
containerupdate: improved error messaging when the file system isn't writable
Fixes https://github.com/tilt-dev/tilt/issues/4206

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
Here's what the error looks like in context:
Screenshot from 2021-02-16 18-20-39

File: internal/containerupdate/exec_updater.go [hunk 1]
```diff
@@ -64,3 +69,15 @@ func (cu *ExecUpdater) UpdateContainer(ctx context.Context, cInfo store.Containe
 
 	return nil
 }
+
+func handleK8sExecError(out *bytes.Buffer, err error) error {
+	msg := strings.ToLower(fmt.Sprintf("%s\n%s", out.String(), err.Error()))
+	if strings.Contains(msg, "permission denied") || strings.Contains(msg, "cannot open") {
+		return fmt.Errorf("denied access to the container filesystem. Please check:\n"+
```
Comment (landism) [inline internal/containerupdate/exec_updater.go:76]: Not a blocker and this is a big improvement over status quo, so feel free to ignore.

I found the sample output a bit confusing:
1) The error doesn't show what was interpreted as "denied access" - I found it confusing to read "denied access to the container filesystem" and then the evidence presented is "original error: command terminated with exit code 2"
2) I really expected the text following "with unexpected error:" to be the error. Can we move the error to the start of this or change the colon to a period?

File: internal/containerupdate/exec_updater.go [hunk 2]
```diff
@@ -64,3 +69,15 @@ func (cu *ExecUpdater) UpdateContainer(ctx context.Context, cInfo store.Containe
...
+			"  1) That the container image has a writable file system\n"+
+			"  2) That the container image default user has write access to the file system\n"+
```
Comment (landism) [inline internal/containerupdate/exec_updater.go:78]: The output in the screenshot appears to be double-spaced, but that doesn't look to be the intent here. Not a blocker, just a bit eyebrow-raising.

**Review by landism — APPROVED** [review]

File: internal/containerupdate/exec_updater.go (hunk 2 above)
Comment (nicks) [reply internal/containerupdate/exec_updater.go:78]: ya, i believe that's a feature of the fallback ui, i'm not entirely sure what the thinking behind this is /cc @hyu

File: internal/containerupdate/exec_updater.go (hunk 1 above)
Comment (nicks) [reply internal/containerupdate/exec_updater.go:76]: ya, i think this is more of a complaint about the way we assemble live-update errors - it's trying to combine errors from a lot of different sources and make sense of them, while also trying not to hide anything from the user. not sure what the right answer is here.

Comment (nicks) [reply internal/containerupdate/exec_updater.go]: here's what it looks like now

Screenshot from 2021-02-17 12-04-11

File: internal/containerupdate/exec_updater.go (hunk 2 above)
Comment (hyu) [reply internal/containerupdate/exec_updater.go]: This design is from a while ago, but I _think_ the intent was to make it very obvious that fallback was happening, and perhaps there were also some bad assumptions about what fallbacks would look like (e.g., not multi-line).

Re-evaluating this now, I think we should just make the fallback message have the same spacing as everything else.

Comment (nicks) [reply internal/containerupdate/exec_updater.go]: adjusting the spacing a bit more, here's what it looks like now:

Screenshot from 2021-02-18 09-57-05

**Pull request merged**